int32 roboyID
int32 control
bool value